src/test/java/org/fogbowcloud/manager/core/models/linkedlist/SynchronizedDoublyLinkedListTest.java

```diff
@@ -0,0 +1,242 @@
+		
+		assertEquals(orderOne, this.list.getCurrent().getOrder());
+		this.list.removeItem(orderOne);
+		assertEquals(orderTwo, this.list.getCurrent().getOrder());
+		assertEquals(orderTwo, this.list.getHead().getOrder());
+		assertNull(this.list.getHead().getPrevious());
+		assertEquals(orderFour, this.list.getTail().getOrder());
+		assertEquals(orderTwo, this.list.getNext());
+		assertEquals(orderThree, this.list.getNext());
+		assertEquals(orderFour, this.list.getNext());
+		assertNull(this.list.getNext());
+	}
+	
+	@Test
+	public void testRemoveItemOnTail() {
+		Order orderOne = createOrder("one"); 
+		Order orderTwo = createOrder("two"); 
+		Order orderThree = createOrder("three");
+
+		this.list.addItem(orderOne);
+		this.list.addItem(orderTwo);
+		this.list.addItem(orderThree);
+
+		assertEquals(orderOne, this.list.getHead().getOrder());
+		assertEquals(orderThree, this.list.getTail().getOrder());
+		
+		this.list.removeItem(orderThree);
+		assertEquals(orderOne, this.list.getHead().getOrder());
+		assertEquals(orderTwo, this.list.getTail().getOrder());
+		assertEquals(orderOne, this.list.getNext());
+		assertEquals(orderTwo, this.list.getNext());
+		assertNull(this.list.getNext());
+	}
+	
+	@Test
+	public void testRemoveItemOneElementOnList() {
+		Order orderOne = createOrder("one"); 
+		
+		this.list.addItem(orderOne);
+		
+		assertEquals(orderOne, this.list.getHead().getOrder());
+		assertEquals(orderOne, this.list.getTail().getOrder());
+		this.list.removeItem(orderOne);
+		assertNull(this.list.getHead());
+		assertNull(this.list.getTail());
+		assertNull(this.list.getNext());
+	}	
+	
+	@Test
+	public void testRemoveItem() throws Exception {
+		Order orderOne = createOrder("one"); 
+		Order orderTwo = createOrder("two"); 
+		Order orderThree = createOrder("three"); 
+		Order orderFour = createOrder("Four"); 
+		
+		this.list.addItem(orderOne);
+		this.list.addItem(orderTwo);
+		this.list.addItem(orderThree);
+		this.list.addItem(orderFour);
+
+		this.list.removeItem(orderThree);
+		assertEquals(orderOne, this.list.getHead().getOrder());
+		assertEquals(orderFour, this.list.getTail().getOrder());
+		assertEquals(orderOne, this.list.getNext());
+		assertEquals(orderTwo, this.list.getNext());
+		assertEquals(orderFour, this.list.getNext());
+		assertNull(this.list.getNext());
+		
+		this.list.resetPointer();
+		this.list.removeItem(orderTwo);
+		assertEquals(orderOne, this.list.getHead().getOrder());
+		assertEquals(orderFour, this.list.getTail().getOrder());
+		assertEquals(orderOne, this.list.getNext());
+		assertEquals(orderFour, this.list.getNext());
+		assertNull(this.list.getNext());
+	}
+
+	@Test
+	public void testReinitializingList(){
+		Order orderOne = createOrder("one");
+		this.list.addItem(orderOne);
+		assertEquals(orderOne, this.list.getHead().getOrder());
+		assertEquals(orderOne, this.list.getCurrent().getOrder());
+		assertEquals(orderOne, this.list.getTail().getOrder());
+		assertEquals(orderOne, this.list.getNext());
```

Comment:
Add an assertNull to check if after the first getNext() a new getNext() would return null.